rice-framework/krad-service-impl/src/main/java/org/kuali/rice/krad/service/impl/DocumentServiceImpl.java

```diff
@@ -211,15 +211,22 @@ private DocumentEvent generateKualiDocumentEvent(Document document,
     @Override
     public Document routeDocument(Document document, String annotation,
             List<AdHocRouteRecipient> adHocRecipients) throws ValidationException, WorkflowException {
+        return routeDocument(document, annotation, null, adHocRecipients);

```

Comment:
The pattern on this method (and all of them that were updated) seems wrong. It seems like instead of passing "null" here for the node name, you should pass document.getAdhocRouteNodeName() which is null by default.